### PR TITLE
Add install feature

### PR DESCRIPTION
**Pull request by julianxhokaxhiu**
Allow third party developers to include this as library, if they need to.

Actually used and required by FF7_Vulkan.

### CODE REVIEW

**Comment by quantumpencil** [comment]
LGTM. @Nax

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+
+file(GLOB_RECURSE SOURCES "src/*.c")
+file(GLOB_RECURSE HEADERS "include/*.h")
+
+add_library(mog STATIC ${HEADERS} ${SOURCES})
 target_include_directories(mog PUBLIC include)
+set_target_properties(mog PROPERTIES PUBLIC_HEADER ${HEADERS})
+
+install(TARGETS mog
+  RUNTIME DESTINATION "bin"
```
Comment (Nax) [inline CMakeLists.txt]: Why not `CMAKE_INSTALL_BINDIR` to be consistent?

Otherwise looks fine :100:

**Review by Nax — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (julianxhokaxhiu) [reply CMakeLists.txt]: Done! Thank you for the suggestion, I totally forgot about it.

**Review by quantumpencil — APPROVED** [review]

**Review by Nax — APPROVED** [review]
:+1:

**Pull request merged**